### PR TITLE
API to return list of tags from a repo

### PR DESCRIPTION
**Pull request by nightfurys**

### CODE REVIEW

**Comment by nightfurys** [comment]
@zhill, can you take a look at the endpoints in particular and suggest better ones? Tried my best to not get the new endpoint to be confused with the existing ones related to resources within the system, not sure if its still ambiguous

File: anchore_engine/clients/docker_registry.py [hunk 1]
```diff
@@ -216,9 +216,12 @@ def get_repo_tags(userId, image_info, registry_creds=None):
...
 
-    lookuptag = 'latest'
-    if image_info['tag']:
-        lookuptag = image_info['tag']
+    if use_lookup_tag:
+        lookuptag = 'latest'
+        if image_info['tag']:
+            lookuptag = image_info['tag']
+    else:
+        lookuptag = None
```
Comment (zhill) [inline anchore_engine/clients/docker_registry.py:224]: This doesn't work in current skopeo implementation if there is no 'latest' tag. See #484

Comment (nightfurys) [reply anchore_engine/clients/docker_registry.py:224]: Ah, I tested it against one of my repos and it worked but the repo had a 'latest' tag. Is that what you mean?

Comment (zhill) [reply anchore_engine/clients/docker_registry.py:224]: Yeah, a repo that doesn't have a 'latest' tag requires that the call include a tag to key off for skopeo. #484 removes that requirement but introduces some regression testing. However, I don't think we should add an API call with a parameter that will quickly be removed. We should probably update skopeo before this change goes in.

Comment (nightfurys) [reply anchore_engine/clients/docker_registry.py:224]: Okay, I've assigned #484 to myself. Will work on that before merging the commits for this issue

**Comment by zhill** [comment]
The current repo-add API call returns metadata that includes the set of tags found. I think this could be easily changed with a ?dry-run=true option that would return the same payload but not persist it. Could omit the creation and update timestamps.

This is an example, but is truncated for brevity:
```
[anchore@anchore-anchore-engine-api]$ anchore-cli --json repo add docker.io/debian
[
    {
        "active": true,
        "created_at": 1594110863,
        "last_updated": 1594110863,
        "record_state_key": "active",
        "record_state_val": null,
        "subscription_id": "197b6ced1fa4e333f9b5f32c646f9f55",
        "subscription_key": "docker.io/debian",
        "subscription_type": "repo_update",
        "subscription_value": "{\"autosubscribe\": true, \"repotags\": [\"10-slim\", \"10.0-slim\", \"10.0\", \"10.1-slim\", \"10.1\", \"10.2-slim\", \"10.2\", \"10.3-slim\", \"10.3\", \"10.4-slim\", \"10.4\", \"10\", \"6.0.10\", \"6.0.8\", \"6.0.9\", \"6.0\", \"6\", \"7-slim\", \"7.10\", \"7.11-slim\", \"7.11\", \"7.3\", \"7.4\", \"7.5\", \"7.6\", \"7.7\", \"7.8\", \"7.9\", \"7\", \"8-slim\", \"8.0\", \"8.1\", \"8.10-slim\", \"8.10\", \"8.11-slim\", \"8.11\", \"8.2\", \"8.3\", \"8.4\", \"8.5\", \"8.6\", \"8.7\", \"8.8\", \"8.9\", \"8\", \"9-slim\", \"9.0\", \"9.1\", \"9.11-slim\", \"9.11\", \"9.12-slim\", \"9.12\", \"9.2\", \"9.3-slim\", \"9.3\", \"9.4-slim\", \"9.4\", \"9.5-slim\", \"9.5\", \"9.6-slim\", \"9.6\", \"9.7-slim\", \"9.7\", \"9.8-slim\", \"9.8\", \"9.9-slim\", \"9.9\", \"9\", \"bullseye-20190708-slim\", \"bullseye-20190708\", \"bullseye-20190812-slim\", \"bullseye-20190812\", \"bullseye-20190910-slim\", \"bullseye-20190910\", \"bullseye-20191014-slim\", \"bullseye-20191014\", \"bullseye-20191118-slim\", \"bullseye-20191118\", \"bullseye-20191224-slim\", \"bullseye-20191224\", \"bullseye-20200130-slim\", \"bullseye-20200130\", \"bullseye-20200224-slim\", \"bullseye-20200224\", \"bullseye-20200327-slim\", \"bullseye-20200327\", \"bullseye-20200414-slim\", \"bullseye-20200414\", \"bullseye-20200422-slim\", \"bullseye-20200422\", \"bullseye-20200511-slim\", \"bullseye-20200511\", \"bullseye-20200514-slim\", \"bullseye-20200514\", \"bullseye-20200607-slim\", \"bullseye-20200607\", \"bullseye-backports\", \"bullseye-slim\", \"bullseye\", \"buster-20170620\", \"buster-20170723\", \"buster-20170907\", \"buster-20171009\", \"buster-20171210\", \"buster-20180213\", \"buster-20180312\", \"buster-20180426\", \"buster-20180625\", \"buster-20180716\", \"buster-20180831\", \"buster-20181011-slim\", \"buster-20181011\", \"buster-20181112-slim\", \"buster-20181112\", \"buster-20181226-slim\", \"buster-20181226\", \"buster-20190122-slim\", \"buster-20190122\", \"buster-20190204-slim\", \"buster-20190204\", \"buster-20190228-slim\", \"buster-20190228\", \"buster-20190326-slim\", \"buster-20190326\", \"buster-20190506-slim\", \"buster-20190506\", \"buster-20190610-slim\", \"buster-20190610\", \"buster-20190708-slim\", \"buster-20190708\", \"buster-20190812-slim\", \"buster-20190812\", \"buster-20190910-slim\", \"buster-20190910\", \"buster-20191014-slim\", \"buster-20191014\", \"buster-20191118-slim\", \"buster-20191118\", \"buster-20191224-slim\", \"buster-20191224\", \"buster-20200130-slim\", \"buster-20200130\", \"buster-20200224-slim\", \"buster-20200224\", \"buster-20200327-slim\", \"buster-20200327\", \"buster-20200414-slim\", \"buster-20200414\", \"buster-20200422-slim\", \"buster-20200422\", \"buster-20200511-slim\", \"buster-20200511\", \"buster-20200514-slim\", \"buster-20200514\", \"buster-20200607-slim\", \"buster-20200607\", \"buster-backports\", \"buster-slim\", \"buster\", \"experimental-20170606\", \"experimental-20170620\", \"experimental-20170723\", \"experimental-20170907\", \"experimental-20171009\", \"experimental-20171210\", \"experimental-20180213\", \"experimental-20180312\", \"experimental-20180426\", \"experimental-20180625\", \"experimental-20180716\", \"experimental-20180831\", \"experimental-20181011\", \"experimental-20181112\", \"experimental-20181226\", \"experimental-20190122\", \"experimental-20190204\", \"experimental-20190228\", \"experimental-20190326\", \"experimental-20190506\", \"experimental-20190610\", \"experimental-20190708\", \"experimental-20190812\", \"experimental-20190910\", \"experimental-20191014\", \"experimental-20191118\", \"experimental-20191224\", \"experimental-20200130\", \"experimental-20200224\", \"experimental-20200327\", \"experimental-20200414\", \"experimental-20200422\", \"experimental-20200511\", \"experimental-20200514\", \"experimental-20200607\", \"experimental\", \"jessie-20170606\", \"jessie-20170620\", \"jessie-20170723\", \"jessie-20170907\", \"jessie-20171009\", \"jessie-20171210\", \"jessie-20180213\", \"jessie-20180312\", \"jessie-20180426\", \"jessie-20180625\", \"jessie-20180716\", \"jessie-20180831\", \"jessie-20181011-slim\", \"jessie-20181011\", \"jessie-20181112-slim\", \"jessie-20181112\", \"jessie-20181226-slim\", \"jessie-20181226\", \"jessie-20190122-slim\", \"jessie-20190122\", \"jessie-20190204-slim\", \"jessie-20190204\", \"jessie-20190228-slim\", \"jessie-20190228\", \"jessie-20190326-slim\", \"jessie-20190326\", \"jessie-20190506-slim\", \"jessie-20190506\", \"jessie-20190610-slim\", \"jessie-20190610\", \"jessie-20190708-slim\", \"jessie-20190708\", \"jessie-20190812-slim\", \"jessie-20190812\", \"jessie-20190910-slim\"},
"userId": "admin"
}
]
```

**Comment by nightfurys** [comment]
Scraped the initial commit and implemented a dry-run extension to repo add API as per @zhill's feedback

**Pull request merged**